File: backend/src/main/java/sn/immosn/backend/favoris/service/FavorisService.java
package sn.immosn.backend.favoris.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import sn.immosn.backend.client.web.favoris.dto.FavorisResponseDto;
import sn.immosn.backend.client.web.favoris.dto.FavorisStatusDto;

import java.util.List;

public interface FavorisService {

    /** Toggle favori — idempotent (ajoute si absent, retire si présent) */
    FavorisStatusDto toggle(Long annonceId, String clientEmail);

    /** Liste paginée des favoris avec détails (pour FavorisView) */
    Page<FavorisResponseDto> getClientFavoris(String clientEmail, Pageable pageable);

    /** Vérifie si une annonce spécifique est en favori */
    FavorisStatusDto checkFavoris(Long annonceId, String clientEmail);

    /**
     * Retourne tous les IDs d'annonces favorites du client.
     * Utilisé par le store Pinia pour pré-charger l'état ❤️ sans limite fixe.
     */
    List<Long> getAllFavorisIds(String clientEmail);
}

File: frontend/src/services/favorisService.js
import api from './api'

export default {
  // POST /api/v1/favoris/{annonceId}/toggle — CLIENT : toggle favori
  toggle(annonceId) {
    return api.post(`/favoris/${annonceId}/toggle`)
  },

  // GET /api/v1/favoris?page&size — liste paginée des favoris avec détails (FavorisView)
  getClientFavoris(page = 0, size = 12) {
    return api.get('/favoris', { params: { page, size } })
  },

  // GET /api/v1/favoris/ids — tous les IDs favoris du client (sans limite fixe)
  // Utilisé par favorisStore pour charger l'état ❤️ au démarrage
  getFavorisIds() {
    return api.get('/favoris/ids')
  },

  // GET /api/v1/favoris/{annonceId}/check — vérifier si une annonce est favorite
  checkFavoris(annonceId) {
    return api.get(`/favoris/${annonceId}/check`)
  },
}
